packages/@aws-cdk/cloudformation-diff/lib/diff/types.ts

```diff
@@ -367,6 +380,10 @@ export class DifferenceCollection<V, T extends IDifference<V>> {
     delete this.diffs[logicalId];
   }
 
+  public set(logicalId: string, diff: T): void {
+    this.diffs[logicalId] = diff;
```

Comment:
Do you think it would be an error if `diffs` already had a `diff` set for a specific `logicalId`? Asking because it could help us catch our own errors easier. So something like: `if (this.diffs.hasOwnProperty(logicalId) => throw an error`. Not sure if this makes sense or not. What do you think?